src/components/molecules/SpaceEditForm/SpaceEditForm.tsx

```diff
@@ -285,16 +285,16 @@ export const SpaceEditForm: React.FC<SpaceEditFormProps> = ({
             ({ id, worldId }) =>
               !(roomVenue?.worldId !== worldId || id === spaceIdFromPortal)
           )
-          .map((venue) => [venue.id, venue])
+          .map((venue) => [venue.slug, venue])
       ),
     [ownedVenues, roomVenue?.worldId, spaceIdFromPortal]
   );
 
   const parentSpace = useMemo(
     () =>
       roomVenue?.parentId
-        ? ownedVenues.find(({ id }) => id === roomVenue?.parentId)
-        : { name: "" },
+        ? ownedVenues.find(({ slug }) => slug === roomVenue?.parentId)
```

Comment:
This doesn't look OK. You can use slugs for user facing concerns, but when it comes down to background logic, it's the IDs that are the steady immutable references.